### PR TITLE
Adjusting max field index setting dynamically for query index

### PR DESCRIPTION
**Pull request by petardz**
Signed-off-by: Petar Dzepina <petar.dzepina@gmail.com>

*Issue #, if available:*

#775

*Description of changes:*

Added adjusting max field index setting dynamically for query index

*CheckList:*
[ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/alerting/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: alerting/src/main/plugin-metadata/plugin-security.policy [hunk 1]
```diff
@@ -5,4 +5,5 @@ grant {
 
     permission java.net.SocketPermission "*", "connect,resolve";
     permission java.net.NetPermission "getProxySelector";
+    permission java.lang.RuntimePermission "accessUserInformation";
```
Comment (lezzago) [inline alerting/src/main/plugin-metadata/plugin-security.policy:8]: Why is this needed?

Comment (petardz) [reply alerting/src/main/plugin-metadata/plugin-security.policy:8]: This is temporary change to try to fix compile issue while rebasing to latest main branch

**Review by lezzago — APPROVED** [review]

**Review by AWSHurneyt — APPROVED** [review]
